# Using Principal Component Analysis (PCA) for Clustering of Metagenomic Sequences


> The objective of this project is to develop a general computational method for characterizing the viral communities in a metagenomic sample inhabiting a certain environment. Our aim is to computationally be able to classify metagenomic DNA sequences into their family groups accurately through finding the optimum size of K-mers to help us calculate the Principal Components Analysis (PCA) for the normalized frequencies of the K-mers. for mode details, please check the report project.

---

### Table of Contents

- [Description](#description)
- [How To Use](#how-to-use)
- [Example](#example)
---

## Description

![A workflow diagram, representing the steps for solving our problem, starting with data collection and preparation, followed by generating  our clustering feature, which is k-mer frequency, followed by using our clustering method, which is the principal component analysis (PCA).](https://github.com/Nourah-Salem/2022-CPBS-Preliminary-Exam-Day-3/blob/main/Images/2022-CPBS-Preliminary-Exam-Day-3%20workflow.png)

#### Technologies

- Technology 1
- Technology 2

[Back To The Top](#read-me-template)

---

## How To Use

#### Installation



#### API Reference

```html
    <p>dummy code</p>
```
[Back To The Top](#read-me-template)

---

## References
[Back To The Top](#read-me-template)

---

## Author Info

- Twitter - [@jamesqquick](https://twitter.com/jamesqquick)
- Website - [James Q Quick](https://jamesqquick.com)

[Back To The Top](#read-me-template)
